### DISCUSSION TITLE
Is there a way to generate a schema file for ruff.toml?

### DISCUSSION THREAD

**Discussion by toxygene** [astral-sh/ruff#14347] [General]
I am working on a tool that gives developers a programatic interface for configuring various tools for projects they work on, but I haven't had much luck generating an interface for the Rust configuration options. It appears that Ruff does have a file containing descriptions of all the options (https://github.com/astral-sh/ruff/blob/ba6c7f6897df6d8020c6b393d1d2f51963fee2b2/crates/ruff_workspace/src/options.rs), there's a tool to build a Markdown file from those descriptions (https://github.com/astral-sh/ruff/blob/a48d779c4e0eb6d9e7bfa11a6226272e4d405bb8/scripts/generate_mkdocs.py#L207), and the settings page on the website (https://docs.astral.sh/ruff/settings/) but none of these are very easy to parse programatically.

Is there any way to convert those options to a JSON Schema or something similar?

**Comment by dhruvmanila** [comment]
We have a JSON schema: https://github.com/astral-sh/ruff/blob/main/ruff.schema.json. Is this what you need?

**Reply by toxygene** [reply]
Brilliant! I'm sorry I didn't find it myself (in the root directory 🤦) before posting this question!

**Reply by dhruvmanila** [reply]
No worries, happy to help :)
